graph.py

```diff
@@ -75,32 +267,56 @@ def formatwhen(name):
             else:
                 return ""
 
+        def formatduration(name):
+            if include_duration:
+                return "\\n{duration}".format(duration=self.get_build_duration(name))
+            else:
+                return ""
+
+        def formatupload(name):
+            if include_upload:
+                return "\\n{upload}".format(upload=self.get_upload_size(name))
+            else:
+                return ""
+
         for base, layers in self.deps.items():
             for layer in layers:
-                txt += "[ {base}{when} ]".format(base=base,
-                                                 when=formatwhen(base))
+                txt += "[ {base}{when}{duration}{upload} ]".format(base=base,
+                                                                   when=formatwhen(base),
+                                                                   duration=formatduration(base),
+                                                                   upload=formatupload(base))
                 txt += " --> "
-                txt += "[ {layer}{when} ]".format(layer=layer,
-                                                  when=formatwhen(layer))
+                txt += "[ {layer}{when}{duration}{upload} ]".format(layer=layer,
+                                                                    when=formatwhen(layer),
+                                                                    duration=formatduration(layer),
+                                                                    upload=formatupload(layer))
                 txt += "\n"
 
         return txt
 
 
-def run(inputfile=None):
+def run(inputfile=None, pulp_base_url=None):
     if inputfile is not None:
         with open(inputfile) as fp:
             builds = json.load(fp)
     else:
         builds = json.load(sys.stdin)
 
-    tree = BuildTree(builds)
+    tree = BuildTree(builds, pulp_base_url)
     tree.trim_excess_tags()
-    print(tree.as_graph_easy_txt(include_datestamp=True))
+    print(tree.as_graph_easy_txt(
+          include_datestamp=True, include_duration=True, include_upload=True))
+
+    (total_duration, total_upload_size, total_layers_size) = tree.calculate_totals()
 
+    sys.stderr.write("Total duration: %s\n" % str(datetime.timedelta(seconds=total_duration)))
+    sys.stderr.write("Total upload size: %s (%s to Pulp and %s to Brew/Koji)\n" % (
+        sizeof_fmt(total_upload_size + total_layers_size),
+        sizeof_fmt(total_layers_size),
+        sizeof_fmt(total_upload_size)))
 
 if __name__ == '__main__':
-    try:
-        run(sys.argv[1])
-    except IndexError:
-        run()
+    args = []
+    if len(sys.argv) > 1:
+        args = [sys.argv[1], sys.argv[2]]
```

Comment:
This doesn't look quite right. If `len(sys.argv)==2`, won't this give an `IndexError`?